sql/dd/impl/bootstrap/bootstrap_ctx.h

```diff
@@ -171,6 +176,10 @@ class DD_bootstrap_ctx {
   bool is_initialize() const {
     return opt_initialize && (m_actual_dd_version == dd::DD_VERSION);
   }
+
+  bool is_system_tablespace_encrypted() {
```

Comment:
Although this works, this doesn't seem right. Server code shouldn't ideally care of InnoDB artifacts.

As i understand, server cannot handle encrypted hardcoded tabelspaces? but it should be done in 'generic' way..

Reply:
Now it is the server which validates the encryption, before it even goes into InnoDB. Thus the encryption needs to be fixed before the server code will validate it.

Reply:
this cannot be an excuse to put InnoDB specific details in server.

SE should convey the intention to upper layer via APIs. and as far as server is concerned it should only care that it is operating on tablespace (and not the specific type).

But as I understand from @georgelorch and @laurynas-biveinis, we are happy to have something 'working'. We can do it post reconciliation. Please create JIRA issue for this.

Reply:
this variable should be named is_mysql_tablespace_encrypted() in the first place - now is is_dd_encrypted() - to not show SE internals - based on that we know if mysql is encrypted from 5.7 during bootstrap.